kpatch-build/create-diff-object.c

```diff
@@ -81,6 +81,60 @@ struct special_section {
  * Functions
  * **********/
 
+static int is_bundleable(struct symbol *sym)
+{
+	if (sym->type == STT_FUNC &&
+	    !strncmp(sym->sec->name, ".text.",6) &&
+	    !strcmp(sym->sec->name + 6, sym->name))
+		return 1;
+
+	if (sym->type == STT_FUNC &&
+	    !strncmp(sym->sec->name, ".text.unlikely.",15) &&
+	    !strcmp(sym->sec->name + 15, sym->name))
+		return 1;
+
+	if (sym->type == STT_OBJECT &&
+	   !strncmp(sym->sec->name, ".data.",6) &&
+	   !strcmp(sym->sec->name + 6, sym->name))
+		return 1;
+
+	if (sym->type == STT_OBJECT &&
+	   !strncmp(sym->sec->name, ".rodata.",8) &&
+	   !strcmp(sym->sec->name + 8, sym->name))
+		return 1;
+
+	if (sym->type == STT_OBJECT &&
+	   !strncmp(sym->sec->name, ".bss.",5) &&
+	   !strcmp(sym->sec->name + 5, sym->name))
+		return 1;
+
+	return 0;
+}
+
+/*
+ * When compiling with -ffunction-sections and -fdata-sections, almost every
+ * symbol gets its own dedicated section.  We call such symbols "bundled"
+ * symbols.  They're indicated by "sym->sec->sym == sym".
+ */
+static void kpatch_bundle_symbols(struct kpatch_elf *kelf)
+{
+	struct symbol *sym;
+
+	list_for_each_entry(sym, &kelf->symbols, list) {
+		if (is_bundleable(sym)) {
+			if (sym->sym.st_value != 0) {
+#ifdef __powerpc__
+				if (!is_localentry_sym(sym))
```

Comment:
For powerpc, I think you also need to move is_localentry_sym() implementation into this file (or prototype in a header file).  Can you do a quick build on power to verify?